GildedRose.cs

```diff
@@ -1,89 +1,115 @@
 
-                if (Items[i].Name != "Sulfuras, Hand of Ragnaros")
-                {
-                    Items[i].SellIn = Items[i].SellIn - 1;
-                }
+            if (item.SellIn > 0)
+            {
+                UpdateQualityBy(item, -1 * decreaseMultiplier);
+            }
+            else
+            {
+                UpdateQualityBy(item, -2 * decreaseMultiplier);
+            }
+        }
 
-                if (Items[i].SellIn < 0)
-                {
-                    if (Items[i].Name != "Aged Brie")
-                    {
-                        if (Items[i].Name != "Backstage passes to a TAFKAL80ETC concert")
-                        {
-                            if (Items[i].Quality > 0)
-                            {
-                                if (Items[i].Name != "Sulfuras, Hand of Ragnaros")
-                                {
-                                    Items[i].Quality = Items[i].Quality - 1;
-                                }
-                            }
-                        }
-                        else
-                        {
-                            Items[i].Quality = Items[i].Quality - Items[i].Quality;
-                        }
-                    }
-                    else
-                    {
-                        if (Items[i].Quality < 50)
-                        {
-                            Items[i].Quality = Items[i].Quality + 1;
-                        }
-                    }
-                }
+        private void UpdateAgedBrieQuality(Item item)
+        {
+            if (item.SellIn > 0)
+            {
+                UpdateQualityBy(item, 1);
+            }
+            else
+            {
+                UpdateQualityBy(item, 2);
+            }
+        }
+
+        private void UpdateBackstagePassesQuality(Item item)
+        {
+            if (item.SellIn > 10)
+            {
+                UpdateQualityBy(item, 1);
+            }
+            else if (item.SellIn > 5)
+            {
+                UpdateQualityBy(item, 2);
+            }
+            else if (item.SellIn > 0)
+            {
+                UpdateQualityBy(item, 3);
+            }
+            else
+            {
+                item.Quality = 0;
+            }
+        }
+
+        private void UpdateSellIn(Item item)
+        {
+            if (item.Name != "Sulfuras, Hand of Ragnaros")
+            {
+                item.SellIn--;
+            }
+        }
+
+        private void UpdateQuality(Item item)
+        {
+            switch (item.Name)
+            {
+                case "Aged Brie":
+                    UpdateAgedBrieQuality(item);
+                    break;
+                case "Backstage passes to a TAFKAL80ETC concert":
+                    UpdateBackstagePassesQuality(item);
+                    break;
+                case "Sulfuras, Hand of Ragnaros":
+                    // Do nothing
```

Comment:
I believe that it's SellIn value does decrease

Reply:
This is inside the method UpdateQuality. It does nothing for Sulfuras with regards to quality. The SellIn property is decreased in the UpdateSellIn method